db/fetcher_test.go

```diff
@@ -183,162 +182,3 @@ func TestFetcherGetAllPrimaryIndexEncodedDocMultiple(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, encdoc)
 }
-
```

Comment:
just want to raise (my usual ;) ) objection to removing some of these unit tests.

Reply:
The function no longer exists, there is nothing to unit test.  FetchNext is covered above.

Reply:
Theres two notable things here in these removed tests that arent covered in the tests that remain. 1) Decoding 2) Value checks. Both of these have value and can be either integrated in to the remaining tests, or we can keep the deleted ones and just update them to use the new APIs

Reply:
> 1) Decoding

If we want unit test for decoding I think they should look quite different to these, and be far more comprehensive - the ones deleted only ever test a single string value, and one low, positive, integer.  I don't think any time would really be saved by converting these tests to decoding unit tests.

> 2) Value checks

That's weird, but you are right, the other tests never actually check the result of `FetchNext`, only that it doesn't error :shrug: - I still very much don't think they are worth keeping though, given that they are offering very little user protection vs our integration tests